phoenix-core/src/main/java/org/apache/phoenix/compile/ScanRanges.java

```diff
@@ -694,4 +694,136 @@ public TimeRange getRowTimestampRange() {
         return rowTimestampRange;
     }
 
+    /**
+     * Produces the list of KeyRanges representing the fully qualified row key by calling into ScanUtil.setKey
+     * repeatedly for every combination of KeyRanges in the ranges field. The bounds will be set according to the
+     * properties of the setKey method.
+     * 
+     * @return list of KeyRanges representing the fully qualified rowkey, coalesced
```

Comment:
@dbwong, is the upper bound of each generated key range is exclusive? If it isn't, can we call nextKey for the upper bound when it is inclusive?

Reply:
Sure.